drivers/UARTSerial.cpp

```diff
@@ -371,38 +371,18 @@ void UARTSerial::disable_tx_irq()
 
 int UARTSerial::enable_input(bool enabled)
 {
-    core_util_critical_section_enter();
-    if (_rx_enabled != enabled) {
-        if (enabled) {
-            UARTSerial::rx_irq();
-            if (!_rxbuf.full()) {
-                enable_rx_irq();
-            }
-        } else {
-            disable_rx_irq();
-        }
-        _rx_enabled = enabled;
-    }
-    core_util_critical_section_exit();
+    api_lock();
+    _enable_input(enabled);
```

Comment:
Once `enable_input` is public in `SerialBase`, we still need to pass this through, as `SerialBase` is private for `UARTSerial`. It deliberately wants to stop people messing with the `SerialBase` without its knowledge.

In this case, `enable_input` is a permitted operation, so we just pass through.

Basically like `set_format`.  Keep the `api_lock` in for consistency, but I'm beginning to doubt its utility. I note it's missing on some calls like `set_baud`.